sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala

```diff
@@ -80,6 +80,54 @@ object EstimationUtils {
     expressions.collect {
       case alias @ Alias(attr: Attribute, _) if attributeStats.contains(attr) =>
         alias.toAttribute -> attributeStats(attr)
+      case alias @ Alias(expn: Expression, _) if isExpressionStatsExist(expn, attributeStats) =>
+        getExpressionStats(alias.toAttribute, expn, attributeStats)
+    }
+  }
+
+  // Support for substring expressions.
+  // TODO: Support for more expressions like Multiply.
```

Comment:
Why do we need to handle individual exprs here? For aggregate stat estimation, we cannot just use upper-bould stat values from a child plan in `AggregateEstimation`?

Reply:
For substring, yes we can pass the child plan stats. But for other expressions it might not be true. For eg: Literal, Concat string etc. We might need to find more expressions if we can estimate stats. For eg: Multiply.

Reply:
Why can we keep discount stats for substring?

Reply:
Sure. I can update stats for substring. Basically max len and avg length can update. Null count remains same. Distinct count will remain upper bound. Actual distinct count can't estimate after applying substring